avro/src/main/scala/magnolify/avro/package.scala

```diff
@@ -18,4 +18,8 @@ package magnolify
 
 package object avro {
   type doc = shared.doc
+
+  case class property(kv: (String, AnyRef))
```

Comment:
Also, if we should add the final modifier if we do not indent to extend the type. 
Otherwise we can provide specialization for known properties and to let user extend the property by using a trait like this

```scala
trait property {
  def key: String
  def value: Anyref
} extends StaticAnnotation with Serializable

object property {
  private case class propertyImpl(key: String, value: String) extends Property
  def apply(key: String, value: String): property
}

case object javaString extened property {
  override def key: String = GenericData.STRING_PROP
  override def value: "String"
}
```